iaso/api/mobile/org_units.py

```diff
@@ -32,52 +40,66 @@ def has_object_permission(self, request, view, obj):
 
 
 class MobileOrgUnitViewSet(viewsets.ViewSet):
-    """Org units API used by the mobile application
+    f"""Org units API used by the mobile application
 
     This API is open to anonymous users for actions that are not org unit-specific (see create method for nuance in
     projects that require authentication).
 
     GET /api/mobile/orgunits/
     POST /api/mobile/orgunits/
+
+    Optionally, {PAGE} and {PAGE_SIZE} parameters can be passed to paginate the results.
+
+    GET /api/mobile/orgunits?{PAGE}=1&{PAGE_SIZE}=100
     """
 
     permission_classes = [HasOrgUnitPermission]
 
     def get_queryset(self):
-        return OrgUnit.objects.filter_for_user_and_app_id(None, self.request.query_params.get("app_id")).filter(
+        return OrgUnit.objects.filter_for_user_and_app_id(None, self.request.query_params.get(APP_ID)).filter(
             validation_status=OrgUnit.VALIDATION_VALID
         )
 
     def list(self, request):
-        queryset = self.get_queryset().prefetch_related("org_unit_type")
+        app_id = self.request.query_params.get(APP_ID)
+        if not app_id:
+            return Response()
+
+        queryset = self.get_queryset().order_by("path").prefetch_related("parent", "org_unit_type")
         queryset = queryset.select_related("org_unit_type")
         response = {}
-        roots = []
-        if request.user.is_authenticated:
-            roots = request.user.iaso_profile.org_units.values_list("id", flat=True)
-        response["roots"] = roots
-
-        app_id = self.request.query_params.get("app_id")
-        if app_id:
-            cached_response = cache.get(app_id)
-        else:
-            return Response()
 
+        pagination = MobileOrgUnitsSetPagination()
+        page = pagination.paginate_queryset(queryset, request)
+        page_number = 1
+        page_size = None
+        if page:
```

Comment:
This works, and you probably found samples like these in the older APIs, but pagination can be fully handled by Django Rest Framework with way less code

Reply:
But maybe this is done in this way to stay backward compatible?

Reply:
I know, and I tried, but for the Django Rest Framework to handle the pagination fully, we have to use a serializer. Since the serialization is done through an `as_dict_for_mobile()` method, I couldn't use a serializer without redoing it all.

Reply:
I think that would be a very small serializer as it does not follow any foreign key
<img width="881" alt="CleanShot 2023-03-09 at 11 09 35@2x" src="https://user-images.githubusercontent.com/185797/223989814-eb701c6c-22f2-42cd-b137-923ddeb2568b.png">
but if some older apps are expecting results without pagination, then, it's probably painful to get right in the DRF (that will paginate everything by default, which is probably a sane approach)